src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/multiples/MultipleReferenceService.java

```diff
@@ -0,0 +1,239 @@
+    public static final String MISSING_MULTIPLE_REFERENCE_ERROR = "The Case (%s) is missing a Multiple Reference";
+
+    private static final String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+    private static final String SEARCH_CASES_FORMAT = "%s/searchCases?ctid=%s";
+    private static final String MULTIPLE_CASE_REFERENCE_KEYWORD = "data.multipleReference.keyword";
+    private static final String ADD_USER_ERROR = "Call to add legal rep to Multiple Case failed for %s";
+
+    public synchronized String createReference(String caseTypeId) {
+        return switch (caseTypeId) {
+            case ENGLANDWALES_BULK_CASE_TYPE_ID -> multipleRefEnglandWalesRepository.ethosMultipleCaseRefGen();
+            case SCOTLAND_BULK_CASE_TYPE_ID -> multipleRefScotlandRepository.ethosMultipleCaseRefGen();
+            default -> throw new IllegalArgumentException(
+                    String.format("Unable to create case reference: unexpected caseTypeId %s", caseTypeId));
+        };
+    }
+
+    public List<String> validateSubcaseIsOfMultiple(CaseData caseData) {
+        List<String> errors = new ArrayList<>();
+        if (!YES.equals(caseData.getMultipleFlag())) {
+            errors.add(String.format(NOT_MULTIPLE_ERROR, caseData.getEthosCaseReference()));
+        }
+        if (caseData.getMultipleReference() == null || caseData.getMultipleReference().isBlank()) {
+            errors.add(String.format(MISSING_MULTIPLE_REFERENCE_ERROR, caseData.getEthosCaseReference()));
+        }
+        return errors;
+    }
+
+    public void addLegalRepToMultiple(CaseDetails caseDetails, String userToAddId) throws IOException {
+        String adminUserToken = adminUserService.getAdminUserToken();
+        String caseType = MultiplesHelper.appendMultipleSuffix(caseDetails.getCaseTypeId());
+        String multipleRef = caseDetails.getCaseData().getMultipleReference();
+        SubmitMultipleEvent multiShell = getMultipleByReference(adminUserToken, caseType, multipleRef);
+
+        String caseId = caseDetails.getCaseId();
+        String multipleId = String.valueOf(multiShell.getCaseId());
+        MultipleData multipleShell = multiShell.getCaseData();
+        if (multipleId.equals("0") || multipleShell == null) {
+            log.warn("Add Respondent Representative to Multiple failed. "
+                    + "Multiple not found for case {}, with MultipleReference {}", caseId, multipleRef);
+            return;
+        }
+
+        String jurisdiction = caseDetails.getJurisdiction();
+        addUserToMultiple(adminUserToken, jurisdiction, caseType, multipleId, userToAddId);
+
+        String caseRef = caseDetails.getCaseData().getEthosCaseReference();
+        updateMultipleLegalRepCollection(
+                adminUserToken, caseType, jurisdiction, multipleShell, multipleId, caseRef, userToAddId);
+    }
+
+    public SubmitMultipleEvent getMultipleByReference(String adminUserToken,
```

Comment:
```suggestion
    private SubmitMultipleEvent getMultipleByReference(String adminUserToken,
```

Reply:
This entire method might also be in et-common

Reply:
I had thought that this might end up getting referenced again since it is such a useful function, but I can change to Private and let the next dev who wants to use it make it Public.